crypto/schnorr-adaptor-signature/types.go

```diff
@@ -0,0 +1,161 @@
+package schnorr_adaptor_signature
+
+import (
+	"bytes"
+	"encoding/hex"
+	"fmt"
+
+	"github.com/btcsuite/btcd/btcec/v2"
+)
+
+const (
+	ModNScalarSize       = 32
+	FieldValSize         = 32
+	JacobianPointSize    = 33
+	AdaptorSignatureSize = JacobianPointSize + ModNScalarSize
+)
+
+type AdaptorSignature struct {
```

Comment:
so the difference of adaptor signature format in block stream spec is missing one `boolean` to indicate whether it needs negation? So the conclusion is that we handle this in vigilante slasher without breaking change to Babylon?

Reply:
> so the difference of adaptor signature format in block stream spec is missing one boolean to indicate whether it needs negation? 

Yes

> So the conclusion is that we handle this in vigilante slasher without breaking change to Babylon?

That's one approach. The other approach (as recommened by Cubist) is to generalise the `NewAdaptorSignatureFromBytes` function such that it handles two formats of the adaptor signatures. I'm working on this

Reply:
Yes, should be pretty easy:

- The 66-byte (old) format contains a point P, scalar s, bool nn. The point P always has even y-coordinate (i.e., the first byte of any valid old-style signature is 2).
- The 65-byte (new) format contains a point P, scalar s.
- P is odd in the new format iff nn is true in the old format.

Reply:
Sounds good, fixed in https://github.com/babylonlabs-io/babylon/commit/6f49e1b813b9e11871831f5d281e16d89c0063a5 and added test cases using our existing BTC delegation data. It works :+1: